src/main/java/com/zaxxer/nuprocess/linux/LibEpoll.java

```diff
@@ -34,21 +34,21 @@
 
    public static native int epoll_ctl(int epfd, int op, int fd, EpollEvent event);
 
-   // technically, events should be an array, but we only ever call with maxevents = 1
```

Comment:
Even to use an array, when using direct binding (`Native.register`) the parameter type would still be `EpollEvent`, _not_ `EpollEvent[]`. Array parameters are not supported with direct binding.

Instead, to allow `maxevents > 1`, you'd do allocate an `EpollEvent[]` using `toArray` and then you'd pass the _first element_ as `events`.